### PR TITLE
Do no check length parameter of ByteBufOutputStream.write(byte[], int, int); method

### PR DESCRIPTION
**Pull request by d-william**
Motivation:

ByteBufOutputStream#write(byte[], int, int); method should respect the contract of the DataOutput interface and of the OutputStream class. In particular, throwing an IndexOutOfBoundsException if offset is negative.
See [DataOutput](https://docs.oracle.com/en/java/javase/21/docs//api/java.base/java/io/DataOutput.html#write(byte%5B%5D,int,int)) and [OutputStream](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/io/OutputStream.html#write(byte%5B%5D,int,int)).
This doesn't happen when the given length is 0. In this case, the method just return.

Modifications:

Removing the check of length parameter preventing enter in buffer.writeBytes(b, off, len);
Add a test, checking the method throws an IndexOutOfBoundsException when offset is negative even if the length parameter is 0.

Result:

The ByteBufOutputStream#write(byte[], int, int); method will throw an IndexOutOfBoundsException when offset is negative even if the length parameter is 0.

### CODE REVIEW

File: buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java [hunk 1]
```diff
@@ -49,6 +49,13 @@ public void testAll() throws Exception {
             // Expected
         }
 
+        try {
+            new ByteBufOutputStream(buf).write(EMPTY_BYTES, -1, 0);
+            fail();
+        } catch (IndexOutOfBoundsException e) {
+            // Expected
+        }
```
Comment (chrisvest) [inline buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java]: You can use `assertThrows()` to make the check simpler

Comment (d-william) [reply buffer/src/test/java/io/netty/buffer/ByteBufStreamTest.java]: Do you want me to change other try/catch checks to assertThrows checks in this test class?
